### PR TITLE
Refactor-18/ Extract static footer menu

### PR DESCRIPTION
**Pull request by trifiasco**
This PR has the following implementations and improvements - 
- Extracts statically wired footer overlay menu.
- Makes it possible to set and pass footer menu from application layer.
- Makes the menu item's cursor to pointer.

Implementation Note - 
1. For now two default values for footer menu is set, `Log Out` and `Switch Theme`. The rest is set and passed from app layer.
2. `Log Out` is wired and should work as expected.
3. `Switch Theme` is not implemented yet, waiting for the theming work to be finished.
4. Icons are not set properly, waiting for the new icon setting workflow to be finished.

### CODE REVIEW

File: framework/PortalFrame/PortalFrame.bbj [hunk 1]
```diff
@@ -70,10 +72,29 @@ class public PortalFrame
             REM btn!.setToolTip(menuitem!.getToolTip())
             REM btn!.setCallback(BBjAPI.ON_BUTTON_PUSH,#this!,"onMenuItemClick")
         wend
+
+        footerMenuItem! = footerMenu!.addMenuItem(footerMenu!.getRoot(),498,"Switch Theme","Switch application theme","WebKit/demo/assets/home.svg")
```
Comment (saadnoor) [inline framework/PortalFrame/PortalFrame.bbj:76]: is it correct image? `WebKit/demo/assets/home.svg`?

File: framework/PortalFrame/PortalFrame.bbj [hunk 2]
```diff
@@ -151,6 +171,12 @@ class public PortalFrame
         item!=#Menu!.getItem(nodeId)
         #navigateTo(item!)
    methodend
+    
+    method public void onFooterMenuItemClick(BBjCustomEvent ev!)
+        nodeId = num(str(ev!.getObject()))
+        item!=#FooterMenu!.getItem(nodeId)
```
Comment (saadnoor) [inline framework/PortalFrame/PortalFrame.bbj:177]: proper spacing needed for readability `item!=#FooterMenu!.getItem(nodeId)`

File: framework/PortalFrame/PortalFrame.bbj [hunk 3]
```diff
@@ -174,6 +200,12 @@ class public PortalFrame
                 #curr_win!.setVisible(1)
             else
                 pgm$ = item!.getProgram()
+                if pgm$ = "void" then
+                    callback$ = item!.getCallback()
+                    x! = null()
```
Comment (saadnoor) [inline framework/PortalFrame/PortalFrame.bbj:205]: do we need this line? `x! = null()`?

File: framework/PortalFrame/PortalFrame.bbj (hunk 1 above)
Comment (trifiasco) [reply framework/PortalFrame/PortalFrame.bbj:76]: No, it's not the correct image. Like I mentioned in the PR's description - 

> Icons are not set properly, waiting for the new icon setting workflow to be finished.

These are example placeholder icon values, will be replaced in a later PR.

File: model/Menu.bbj [hunk 4]
```diff
@@ -97,6 +98,14 @@ class public MenuItem
         methodret #Program$
     methodend
 
+    method public void setCallback(BBjString callback$)
```
[the remark this answers is not in the document]
Comment (trifiasco) [reply model/Menu.bbj:101]: I kept the method `setCallback` to keep it consistent with the setting callback method in BBjAPI. But I agree in this case property name and method name were inconsistent. So I changed the property name from `CallbackName$` to `Callback$`.

File: widgets/Drawer/components/DrawerFooter/DrawerFooter.bbj [hunk 5]
```diff
@@ -93,32 +100,30 @@ class public DrawerFooter extends BBjWidget
...
-        logoutIcon!.addStyle("drawerFooterOverlayLogoutIcon")
+
+        declare DrawerMenuTileEntry entry!
+        declare BBjChildWindow spacer!
+        templateRows! = ""
         
-        logoutText! = logout!.addStaticText(logout!.getAvailableControlID(),0,0,0,0,"Log Out")
-        logoutText!.addStyle("drawerFooterPopUpTextStyle")
-        logoutText!.setStyle("grid-column-start","2")
+        for i = 0 to #tiles!.size() -1
```
[the remark this answers is not in the document]
Comment (trifiasco) [reply widgets/Drawer/components/DrawerFooter/DrawerFooter.bbj:108]: I don't see any necessity for creating an extra method. Firstly, these codes are not being reused elsewhere. Also `createOverlay` method is respecting `Single responsibility Principle`, that is - it's creating the overlay by rendering the components shown in the overlay. The above code is part of rendering the overlay.

[the remark this answers is not in the document]
Comment (saadnoor) [reply widgets/Drawer/components/DrawerFooter/DrawerFooter.bbj:108]: My point was to improve readability of the code.  If we could move the codes related to templateRow generating to another function, then the `createOverlay` function only has 2 task: modifying `overlay` and modifying `contentWrapper`, in that case code would be more readable and self documented. 

But I agree with you, as generating templateRow functionality isn't being reused, we can skip this. Thanks!

**Pull request merged**